lib/ui/painting.dart

```diff
@@ -1567,6 +1567,22 @@ class FrameInfo extends NativeFieldWrapperClass2 {
   Image get image native 'FrameInfo_image';
 }
 
+/// The default max amount of memory to use for caching decoded animated image
+/// frames compared to total undecoded size. See [decodedCacheRatioCapOverride].
+const double kDefaultDecodedCacheRatioCap = 25.0;
+
+/// By default individual frames of animated images are cached into memory to
+/// avoid using CPU to re-decode them for every loop in the animation. This
+/// behavior will result in out of memory crashes when decoding large
+/// (or large amounts of) animated images. Set this value to limit how much
+/// memory each animated image is allowed to use to cache decoded frames
+/// compared to its compressed size. For example, setting this to `2.0` means
+/// that a 400KB GIF would be allowed at most to use 800KB of memory caching
+/// unessential decoded frames. A setting of `1.0` or less disables all caching
+/// of unessential decoded frames. See [kDefaultDecodedCacheRatioCap] for the
+/// default value.
```

Comment:
the first paragraph of dartdocs needs to be a short sentence since it's used in various places as a standin for the rest. Add a paragraph like "Specifies the maximum multiple of the compressed image size to use when caching an animated image." or some such.